src/runtime/micro/device/utvm_runtime.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ *  Copyright (c) 2019 by Contributors
+ * \file utvm_runtime.cc
+ * \brief micro device init stub
+ */
+#include "utvm_runtime.h"
+
+// Task pointers must be patched before calling a function.
+UTVMTask task;
+
+// These pointers are patched at load time to point to the workspace section.
+char *utvm_workspace_begin = NULL;  // NOLINT(*)
```

Comment:
consider making the global state a struct. UTVMContext

Reply:
I'm not sure if we can do this in a clean way, because we rely on the symbols being exposed at the global level to be able to read from and write to them.  If it's some `utvm_context` struct being exposed, then we would need to manually calculate the offset of `workspace_begin` to patch that pointer.  I'm hesitant to add any more fragility, because we still don't have a great solution for maintaining a correspondence between struct padding on the device side and the host side.  Let me know what you think.